core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -37,6 +38,19 @@
      * default path allows the Axis M1011 cameras used in FRC to work when only an IP address is supplied.
      */
     public final static String DEFAULT_IP_CAMERA_PATH = "/mjpg/video.mjpg";
+    private static final int
+            /**
+             * Connecting to a device can take the most time.
+             * This should have a little bit of leeway
+             */
+            IP_CAMERA_CONNECTION_TIMEOUT = 14;
```

Comment:
I would just do 5 for the read timeout.  In real life, people are gonna connect to axis cameras on an ethernet link, or maybe sometimes on the safe WiFi network.

The connection timeout should be unnecessary since we just retry the same connection when the timeout is up.  If there's a problem that can't be solved within the timeout (like the hostname lookup failed), it would fail immediately anyways.

IMO this is really self-documenting (too much documentation for simple things *is* a bad sign)